tests/unit/package_managers/yarn_classic/test_resolver.py

```diff
@@ -0,0 +1,246 @@
+import re
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from pyarn.lockfile import Package as PYarnPackage
+
+from cachi2.core.errors import PackageRejected, UnexpectedFormat
+from cachi2.core.package_managers.yarn_classic.resolver import (
+    FilePackage,
+    GitPackage,
+    LinkPackage,
+    RegistryPackage,
+    UrlPackage,
+    YarnClassicPackage,
+    _classify_pyarn_package,
+    _get_packages_from_lockfile,
+    _is_from_npm_registry,
+    _is_git_url,
+    _is_tarball_url,
+    resolve_packages,
+)
+from cachi2.core.rooted_path import PathOutsideRoot, RootedPath
+
+
+@pytest.mark.parametrize(
+    "url, expected_output",
+    [
+        ("https://foo.com/bar.tar.gz", True),
+        ("https://foo.com/bar.tgz", True),
+        ("https://foo.com/bar.tar", True),
+        ("http://foo.com/bar.tar.gz", True),
+        ("http://foo.com/bar.tgz", True),
+        ("http://foo.com/bar.tar", True),
+        ("https://codeload.github.com/org/foo/tar.gz/fffffff", True),
+        ("ftp://foo.com/bar.tar", False),
+        ("git+https://git.host.com/some/path", False),
+        ("https://foo.com/bar", False),
+        ("https://foo.com/bar.txt", False),
+    ],
+)
+def test_is_tarball_url(url: str, expected_output: bool) -> None:
```

Comment:
What do you think about splitting the test in two: a positive and a negative test?

```python
@mark.parametrize("url",
 ["https://foo.com/bar.tgz", ...
 ],
)
def test__is_tarball_url_can_parse_correct_tarball_urls(url):
    assert _is_tarball_url(url)

@mark.parametrize("url",
 ["https://foo.com/bar.tar", ...
 ],
)
def test__is_tarball_url_rejects_incorrect_tarball_urls(url):
    assert not _is_tarball_url(url)
```